client/vaultclient/vaultclient.go

```diff
@@ -341,13 +346,16 @@ func (c *vaultClient) GetConsulACL(token, path string) (*vaultapi.Secret, error)
 	return c.client.Logical().Read(path)
 }
 
-// RenewToken renews the supplied token for a given duration (in seconds) and
-// adds it to the min-heap so that it is renewed periodically by the renewal
-// loop. Any error returned during renewal will be written to a buffered
-// channel and the channel is returned instead of an actual error. This helps
-// the caller be notified of a renewal failure asynchronously for appropriate
-// actions to be taken. The caller of this function need not have to close the
-// error channel.
+// RenewToken pushes the supplied token to the min-heap for an immediate
+// renewal for a given duration (in seconds) and blocks until the renewal loop
+// has processed it. The token is then renewed periodically until Stop() or
+// StopRenewToken() is called.
+//
+// Any error returned during the periodical renewal will be written to a
+// buffered channel and the channel is returned instead of an actual error.
+// This helps the caller be notified of a renewal failure asynchronously for
+// appropriate actions to be taken. The caller of this function need not have
```

Comment:
This part of the comment already existed, so I just kept it as is, but reading again it's indeed weirdly phrased 😅 

It's also not necessary because `RenewToken` returns a read channel and trying to close it is a compilation error: https://go.dev/play/p/MQxK_XyGa-7

I will remove this last sentence.